Add edit text example

lord opened the pull request:
It's pretty slow and the code is kind of crappy, but I think it's good enough for review. Having a text editing example is I think important to make sure our text editing APIs don't degrade.

Chinese text input no longer shows the composition window; still have to look into why that is, but perhaps I can fix that in another PR.

xStrom commented:
Clippy errors were resolved with #77 so you can rebase for CI success.

lord commented:
(rebase complete, this should be ready for review)

jneem commented on examples/edit_text.rs:
```diff
@@ -0,0 +1,509 @@
+    let app = Application::new().unwrap();
+    let mut window_builder = glazier::WindowBuilder::new(app.clone());
+    window_builder.resizable(true);
+    window_builder.set_size((WIDTH as f64 / 2., HEIGHT as f64 / 2.).into());
```
Why divide by 2?

lord replied:
This is another copy from `shello.rs` :/

jneem commented on examples/edit_text.rs:
```diff
@@ -0,0 +1,509 @@
+
+impl PartialEq<ParleyBrush> for ParleyBrush {
+    fn eq(&self, _other: &ParleyBrush) -> bool {
+        true // FIXME
```
What's this?

lord replied:
Copied over from the `shello.rs` example :/

dfrg replied:
`peniko::Brush` is `PartialEq` and has been for a while so we should be able to just derive here now. I think this was a limitation in old piet-gpu code. We may need to update some revs for this to be visible.

jneem reviewed (approved):
Looks good! I had a couple of silly little questions, but my main question is: what's the deal with the `#[non_exhaustive]` on `HitTestPoint`? I assume it was originally put there for a reason, but I'm missing a bunch of context here...

lord commented:
Thanks for reviewing! the `non_exhaustive` was something I put there in my last PR; I'm pretty sure I blindly copied it from another struct without thinking.

jneem commented:
Ah, gotcha. I noticed that it's present in piet, so I figured that's where it came from.

lord commented:
Oop, you're totally right! I must have copied it from there. Looks like in piet this was possible because there was a constructor on the type, so I've gone ahead and added back in the `non_exhaustive` annotation and added a constructor.
